actions/cis-benchmark

```diff
@@ -0,0 +1,120 @@
+#!/usr/local/sbin/charm-env python3
+import os
+import shlex
+import subprocess
+import sys
+import tempfile
+
+from pathlib import Path
+from shutil import which
+
+from charmhelpers.core import hookenv
+from charmhelpers.fetch.giturl import GitUrlFetchHandler
+from charms.layer import snap
+from charms.reactive import is_flag_set
+
+
+BENCH_CMD = '/usr/local/bin/kube-bench'
+BENCH_PKG = 'github.com/aquasecurity/kube-bench'
+
+
+def fail(msg):
+    hookenv.action_fail(msg)
+    sys.exit()
+
+
+def install():
+    '''Install kube-bench and related configuration.
+
+    Install go if needed. Pull, build, and setup snap-based benchmark
+    configuration.
+    '''
+    go_bin = which('go')
+    if not go_bin:
+        go_bin = '/snap/bin/go'
+        snap.install('go', channel='stable', classic=True)
+
+    go_cache = os.getenv('GOCACHE', '/var/snap/go/common/cache')
+    go_path = os.getenv('GOPATH', '/var/snap/go/common')
+    env = os.environ.copy()
+    env['GOCACHE'] = go_cache
+    env['GOPATH'] = go_path
+    Path(go_path).mkdir(parents=True, exist_ok=True)
+
+    go_cmd = ('{bin} get {pkg} '
+              'github.com/golang/dep/cmd/dep'.format(bin=go_bin, pkg=BENCH_PKG))
+    try:
+        subprocess.check_call(shlex.split(go_cmd), cwd=go_path, env=env)
+    except subprocess.CalledProcessError:
+        fail('Failed to run: {}'.format(go_cmd))
+
+    go_cmd = ('{bin} build -o {out} {base}/src/{pkg}'.format(
+        bin=go_bin, out=BENCH_CMD, base=go_path, pkg=BENCH_PKG))
+    try:
+        subprocess.check_call(shlex.split(go_cmd), cwd=go_path, env=env)
+    except subprocess.CalledProcessError:
+        fail('Failed to run: {}'.format(go_cmd))
+
+    cfg_dir = '{bin}/src/{pkg}/cfg-ck'.format(bin=go_path, pkg=BENCH_PKG)
+    GitUrlFetchHandler().clone('https://github.com/charmed-kubernetes/kube-bench-config.git',
+                               dest=cfg_dir)
+
+
+def run():
+    app = hookenv.charm_name() or ''
+    version = '1.13-snap-k8s'
```

Comment:
1.13? Do we want to make this an argument?

Reply:
No, these are hard coded subdirs defined [upstream](https://github.com/aquasecurity/kube-bench/tree/master/cfg), which we mirror and tweak in our [custom config repo](https://github.com/charmed-kubernetes/kube-bench-config/). IMO, it's better for the action to figure out the right subdir rather than leave it up to the user.